codebase-pipelines/tests/unit.tftest.hcl

```diff
@@ -28,11 +28,58 @@ override_data {
   }
 }
 
+override_data {
+  target = data.aws_iam_policy_document.assume_codepipeline_role
+  values = {
+    json = "{\"Sid\": \"AssumeCodepipelineRole\"}"
+  }
+}
+
+override_data {
+  target = data.aws_iam_policy_document.ecs_access_for_codebuild_manifests
+  values = {
+    json = "{\"Sid\": \"CodeBuildDeployManifestECS\"}"
+  }
+}
+
+override_data {
+  target = data.aws_iam_policy_document.ecs_deploy_access_for_codebase_pipeline
+  values = {
+    json = "{\"Sid\": \"CodePipelineECSDeploy\"}"
+  }
+}
+
+override_data {
+  target = data.aws_iam_policy_document.assume_event_bridge_policy
+  values = {
+    json = "{\"Sid\": \"AssumeEventBridge\"}"
+  }
+}
+
+override_data {
+  target = data.aws_iam_policy_document.event_bridge_pipeline_trigger
+  values = {
+    json = "{\"Sid\": \"EventBridgePipelineTrigger\"}"
+  }
+}
```

Comment:
I know we cannot assert on the contents on a plan, but we should be able to assert on some basic things, like have we at least used the correct policy document...

https://github.com/uktrade/terraform-platform-modules/commit/1cd29ab7f4955166656306d4a3086f707ebf314d#diff-8494f4b8e2b47c589035d642f9c28ffa867e9c9a25dc0055f6b9fc4928796860

Reply:
I realised we can test more stuff. It's only policy documents with resources that need to be evaluated at apply that we can't tests.

So I've added a bunch of policy document tests and links between policies and roles where possible.